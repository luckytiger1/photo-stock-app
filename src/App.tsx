import React, { lazy, Suspense } from 'react';
import { Switch, Route } from 'react-router-dom';
import Header from './components/Header/Header';
import Spinner from './components/Spinner/Spinner';

const MainPage = lazy(() => import('./components/MainPage/MainPage'));
const Favorites = lazy(() => import('./components/Favorites/Favorites'));
const HistoryPage = lazy(() => import('./components/HistoryPage/HistoryPage'));
const PhotoPageContainer = lazy(() =>
  import('./components/PhotoPage/PhotoPageContainer'),
);

const App = () => {
  return (
    <div className="App ">
      <Suspense fallback={<Spinner />}>
        <Header />
        <Switch>
          <Route exact path="/" component={MainPage} />
          <Route path="/favorites" component={Favorites} />
          <Route path="/history" component={HistoryPage} />
          <Route path="/:photoId" component={PhotoPageContainer} />
        </Switch>
      </Suspense>
    </div>
  );
};

export default App;
