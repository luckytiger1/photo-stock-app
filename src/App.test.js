import React from 'react';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import App from './App';

it('<Favorites /> renders correctly', () => {
  expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
  expect(shallow(<App />)).toMatchSnapshot();
});
