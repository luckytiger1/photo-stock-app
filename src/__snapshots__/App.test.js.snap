// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorites /> renders correctly 1`] = `
<div
  className="App "
>
  <Suspense
    fallback={<Spinner />}
  >
    <withRouter(Header) />
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        path="/favorites"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        path="/history"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        path="/:photoId"
      />
    </Switch>
  </Suspense>
</div>
`;
